src/components/logger/response-time-log.ts

```diff
@@ -0,0 +1,75 @@
+ * MIT License                                                                    *
+ *                                                                                *
+ * Copyright (c) 2021 Hyperjump Technology                                        *
+ *                                                                                *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy   *
+ * of this software and associated documentation files (the "Software"), to deal  *
+ * in the Software without restriction, including without limitation the rights   *
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell      *
+ * copies of the Software, and to permit persons to whom the Software is          *
+ * furnished to do so, subject to the following conditions:                       *
+ *                                                                                *
+ * The above copyright notice and this permission notice shall be included in all *
+ * copies or substantial portions of the Software.                                *
+ *                                                                                *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR     *
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,       *
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE    *
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER         *
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,  *
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE  *
+ * SOFTWARE.                                                                      *
+ **********************************************************************************/
+
+import { differenceInHours } from 'date-fns'
+import { ProbeRequestResponse } from '../../interfaces/request'
+
+// global variable
+// we can say it is safe to use global variable here since the probe is processed sequentially as mentioned in probe/index.ts line 143: intentionally wait for a request to finish before processing next request in loop
+let startTime24HourCycle = new Date()
+let responseCount = 0
+let totalResponseTime = 0
+export let maxResponseTime = 0
+export let minResponseTime = 0
+export let averageResponseTime = 0
+
+function resetlogs() {
+  startTime24HourCycle = new Date()
+  responseCount = 0
+  totalResponseTime = 0
+  maxResponseTime = 0
+  minResponseTime = 0
+  averageResponseTime = 0
+}
+
+function checkIs24HourHasPassed() {
+  const now = new Date()
+  const diffInHours = differenceInHours(now, startTime24HourCycle)
+  if (diffInHours > 24) {
+    return true
+  }
+
+  return false
+}
+
+export function logResponseTime(probeRes: ProbeRequestResponse) {
+  if (checkIs24HourHasPassed()) {
+    resetlogs()
+  }
```

Comment:
Maybe adding to @nicnocquee comment https://github.com/hyperjumptech/monika/pull/442#discussion_r716997907

The default for status-notification schedule is 06:00 AM

For instance I start monika on Monday 05:30 AM, then 30 minutes later I will receive summary from 05:30 to 06:00 AM. That's fine for the first day.

The next day Tuesday at 05:30 logs will be reset. Then 30 minutes later monika sends status notification. But now what I receive is only summary from 05:30 to 06:00, not the last 24 hours.

Reply:
ah nice catch. So it's better to reset logs after sending the notification, no need checking the 24 hour

Reply:
Resetting the logs after sending the notification also lead to unexpected behavior.

For example when you set monika to send notification every 30 minutes

```sh
monika --status-notification "*/30 * * * *"
```

resetting the logs after sending notification will cause the summary to be valid only for the last 30 minutes, not 24 hours.

Reply:
I see. I'll have to think about it first

Reply:
fixed, please recheck